packages/starknet-snap/test/src/executeTxn.test.ts

```diff
@@ -58,7 +82,39 @@ describe('Test function: executeTxn', function () {
     apiParams.requestParams = requestObject;
   });
 
-  it('should executeTxn correctly', async function () {
+  it('should executeTxn correctly and deploy account if not deployed', async function () {
+    sandbox.stub(utils, 'isAccountDeployed').resolves(false);
+    const stub = sandbox.stub(utils, 'executeTxn').resolves({
+      transaction_hash: 'transaction_hash',
+    });
+    const result = await executeTxn(apiParams);
+    const { privateKey } = await utils.getKeysFromAddress(
+      apiParams.keyDeriver,
+      STARKNET_MAINNET_NETWORK,
+      state,
+      account1.address,
+    );
+
+    expect(result).to.eql({
+      transaction_hash: 'transaction_hash',
+    });
+    expect(stub).to.have.been.calledOnce;
+    expect(stub).to.have.been.calledWith(
+      STARKNET_MAINNET_NETWORK,
+      account1.address,
+      privateKey,
+      {
+        entrypoint: 'transfer',
+        calldata: ['0', '0', '0'],
+        contractAddress: createAccountProxyTxn.contractAddress,
+      },
+      undefined,
+      { maxFee: '22702500105945', nonce: 1 },
+    );
```

Comment:
```suggestion
    const stub = sandbox.stub(utils, 'executeTxn').resolves({
      transaction_hash: 'transaction_hash',
    });
    const createAccountStub = sandbox.stub(createAccountUtils, 'createAccount').resolvesThis();
    const result = await executeTxn(apiParams);
    const { privateKey } = await utils.getKeysFromAddress(
      apiParams.keyDeriver,
      STARKNET_MAINNET_NETWORK,
      state,
      account1.address,
    );

    expect(result).to.eql({
      transaction_hash: 'transaction_hash',
    });
    expect(stub).to.have.been.calledOnceWith(
      STARKNET_MAINNET_NETWORK,
      account1.address,
      privateKey,
      {
        entrypoint: 'transfer',
        calldata: ['0', '0', '0'],
        contractAddress: createAccountProxyTxn.contractAddress,
      },
      undefined,
      { maxFee: '22702500105945', nonce: 1 },
    );
    expect(createAccountStub).to.have.been.calledOnceWith(sinon.match.any, true, true);
```

we check if createAccountStub has triggered or not

remember to include import * as createAccountUtils from '../../src/createAccount'; at the top if the file